Add support for Wagtail 5 and 6

wsot opened the pull request:
Wagtail 4 deprecated importing `blocks` and `hooks` from `wagtail.core`, and removal occured in Wagtail 5.
Wagtail 5.1 deprecated the hook `insert_editor_css`, and the hook was removed in Wagtail 6.

This replaces the use of deprecated imports and hooks with the suggested replacements from Wagtail.

Closes #14

jasondilworth56 committed:
Update import to reflect changes in Wagtail

wagtail.core.* imports were changed to wagtail.* in 3.0 and this line now fails in 5.0

wsot committed:
Retain backward compatibility to Wagtail <4.0

Import from `wagtail.core` was deprecated in Wagtail 4.0 and removed in
5.0. This adds a version check to import `blocks` and `hooks` from
`wagtail` rather than `wagtail.core` for Wagtail 4.0 and later.

wsot committed:
Replace use of removed `insert_editor_css`

The hook `insert_editor_css` was deprecated in Wagtail 5.1 and removed
in Wagtail 6.0. This replaces `insert_editor_css` with
`insert_global_admin_css` for Wagtail 5.0 and later, is this release
is the first with Wagtail >=5.0 compatibility.

G-Rath reviewed (approved):
squash when merging

---

looks good to me!

wsot committed:
Add additional classifiers to setup.py

Add additional classifiers to represent current supported platforms.

wsot committed:
Update Copyright dates in README.md to be current

jadedarko commented on wagtailcolumnblocks/blocks.py:
```diff
@@ -7,16 +7,27 @@
+    from wagtail import blocks, hooks
+
+    @hooks.register('insert_global_admin_css')
+    def global_admin_css():
```
**suggestion**: because this function is the same in both cases, it might be worth having outside the conditional blocks & conditionally decorating it

wsot replied:
~I have a (weak) preference for making blocks like this such that each chunk is standalone, so if you _only_ want to support 5.0 and later, you just delete the version check and the `else` block.~

~I'm not terribly attached to it though.~

Edit: Going to change to the first of the two below.

Were you thinking one of these two alternatives below? Or something else?
```
if wagtail_version >= (5, 0):
    from wagtail import blocks, hooks
    hook_register_name = 'insert_global_admin_css'

else:
    from wagtail.core import blocks, hooks
    hook_register_name = 'insert_editor_css'


@hooks.register(hook_register_name)
def global_admin_css():
    return format_html(
        '<link rel="stylesheet" href="{}">',
        static('wagtailcolumnblocks/columns.css')
    )
```

or

```
if wagtail_version >= (5, 0):
    from wagtail import blocks, hooks
    hook_decorator_fn = @hooks.register('insert_global_admin_css')

else:
    from wagtail.core import blocks, hooks
    hook_decorator_fn = @hooks.register('insert_editor_css')


@hook_decorator_fn
def global_admin_css():
    return format_html(
        '<link rel="stylesheet" href="{}">',
        static('wagtailcolumnblocks/columns.css')
    )
```

jonholdsworth commented:
Excellent.

wsot committed:
Move common logic out of version check

Moves logic shared between both old and new Wagtail
version support out of a conditional block to improve
visibility of the specific version-supporting change.
